bids/analysis/base.py

```diff
@@ -0,0 +1,366 @@
+import json
+from bids.grabbids import BIDSLayout
+from .variables import (load_variables, SimpleColumn, BIDSVariableCollection,
+                        merge_collections)
+from . import transform
+from collections import namedtuple
+from six import string_types
+import pandas as pd
+import numpy as np
+
+
+DesignMatrix = namedtuple('DesignMatrix', ('data', 'entities'))
+
+
+class Analysis(object):
+
+    ''' Represents an entire BIDS-Model analysis.
+    Args:
+        layout (BIDSLayout, str): A BIDSLayout instance or path to pass on
+            to the BIDSLayout initializer.
+        model (str or dict): a BIDS model specification. Can either be a
+            string giving the path of the JSON model spec, or an already-loaded
+            dict containing the model info.
+        variables (dict): An optional dictionary containing all variables to
+            use. Keys are level names ('time', 'run', 'session', or 'subject');
+            values are BIDSVariableCollections. If None, all variables will be
+            read from the current layout. Mainly useful for cases where custom
+            arguments need to be passed during variable-reading.
+    '''
+
+    def __init__(self, layout, model, variables=None, **selectors):
+
+        self.variables = variables
+        self.selectors = selectors
+
+        if not isinstance(layout, BIDSLayout):
+            layout = BIDSLayout(layout)
```

Comment:
So if I'm understanding this correctly, Analysis will make no attempt to merge layouts, and it's the user's responsibility to provide a merged layout if something besides a simple layout is desired? Or will there be an alternative way to specify multiple base directories?

Reply:
No, it should work okay... the latest release of grabbit allows passing multiple base directories to `Layout()`. Internally, each one is initialized separately, and then they're merged together and returned as a single one. The `BIDSLayout` initializer just passes `path` through to `Layout`, so it should all work as expected--though admittedly I haven't verified that (feel free to).

That does leave unaddressed the issue @adelavega raised in #101, where you might want to have extra event files or other variable files in some derivate directory, but at present these might be ignored. But that should be straightforward to patch (e.g., by providing an argument to `load_variables` that enables loading of multiple files that look redundant).